vpn-ca/main.go

```diff
@@ -147,12 +153,18 @@ func getTemplate(commonName string, notAfter time.Time, keyUsage x509.KeyUsage,
 	}
 }
 
-func sign(caInfo *caInfo, commonName string, tpl *x509.Certificate) *x509.Certificate {
-	key := generateKey(filepath.Join(caInfo.caDir, fmt.Sprintf("%s.key", commonName)))
+func sign(caInfo *caInfo, commonName string, tpl *x509.Certificate, signType string) *x509.Certificate {
+	absPath, _ := filepath.Abs(caInfo.caDir)
+	signPath := filepath.Join(filepath.Dir(absPath), signType)
+	if _, err := os.Stat(signPath); os.IsNotExist(err) {
+		os.Mkdir(signPath, os.ModePerm)
```

Comment:
os.ModePerm is 0777 for file permissions? that is a bit too lenient, maybe 0644 would be better default?

Reply:
changed to 0644.
would 0600 be even better?, since no other users should be authorized to view private cert/keys

Reply:
0700 would be best :)

Reply:
Updated to 0700